### PR TITLE
[SYCL][XPTI] Fix static analysis tool warnings

### PR DESCRIPTION
**Pull request by alexbatashev**
Make sure nullptr is not dereferenced accidentally.

### CODE REVIEW

**Review by s-kanaev — DISMISSED** [review]
LGTM

**Review by vladimirlaz — DISMISSED** [review]

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 1]
```diff
@@ -528,6 +528,7 @@ RT::PiProgram ProgramManager::getBuiltPIProgram(
       std::make_pair(std::make_pair(std::move(SpecConsts), KSId),
                      std::make_pair(PiDevice, CompileOpts + LinkOpts)),
       AcquireF, GetF, BuildF);
+  assert(BuildResult != nullptr && "Invalid build result");
```
[the remark this answers is not in the document]
Comment (alexbatashev) [reply sycl/source/detail/program_manager/program_manager.cpp:531]: In CI we always build and test with assertions. Honestly, I don’t see a code path, that returns nullptr. And I assume assert makes more sense here.

[the remark this answers is not in the document]
Comment (tovinkere) [reply sycl/source/detail/program_manager/program_manager.cpp:531]: @alexbatashev If BuildResult is guaranteed to not be nullptr, can we document it somewhere that this is the case, hence the use of assert()?

[the remark this answers is not in the document]
Comment (alexbatashev) [reply sycl/source/detail/program_manager/program_manager.cpp:531]: @tovinkere I can add a comment to getOrBuild
Generally, I think this is a perfect case for C++ [contracts](http://open-std.org/JTC1/SC22/WG21/docs/papers/2021/p2388r4.html), but they have not been standardized yet, unfortunately.

**Review by vladimirlaz — APPROVED** [review]

**Review by tovinkere — APPROVED** [review]
LGTM

**Pull request merged**